### PR TITLE
teams listing

### LINKED ISSUE #1483: provide a way to browse/search teams
I want to be able to easily find teams to contribute to.

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/950213-gittip-com-teams?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F85909&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by whit537**
Closes #1483.

### CODE REVIEW

**Comment by Changaco** [comment]
Sooner or later we're going to need pagination for the teams listing.

File: www/explore/teams/index.spt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+         , p.receiving
+         , p.receiving / t.nmembers AS dollars_per_member 
+      FROM ( SELECT team            AS name
+                  , count(member)   AS nmembers
+               FROM current_takes
+           GROUP BY team
+            ) AS t
+      JOIN participants p 
+        ON p.username = t.name
+  ORDER BY dollars_per_member DESC, nmembers ASC
```
Comment (Changaco) [inline www/explore/teams/index.spt]: Since the focus is on receiving we should filter out the teams that don't accept tips.

Comment (whit537) [inline www/explore/teams/index.spt]: Hmm ... we allow such a thing? Okay.

**Comment by Changaco** [comment]
It looks weird to have a completely empty sidebar.

File: www/explore/teams/index.spt (hunk 1 above)
Comment (whit537) [inline www/explore/teams/index.spt]: Done in 1c3bc7c77c749b1068828706a4bac3471dd62bc7. I also added `goal` and `% funded` columns:

screen shot 2015-02-13 at 4 00 27 pm

Comment (Changaco) [inline www/explore/teams/index.spt]: You put the goals in the receiving column, and vice versa.

**Comment by whit537** [comment]
> It should be `goal >= 0`.

Why? When goal is 0 they are a patron. The primary purpose of this page is to enable individuals to find teams that they could join on Gratipay to do work and take funding. If we're going to preclude `goal < 0` then we should preclude `goal = 0` as well.

**Comment by whit537** [comment]
And we definitely should *include* `goal is null`.

**Comment by whit537** [comment]
(But maybe you weren't suggesting we drop the `OR p.goal IS NULL`.)

**Comment by whit537** [comment]
> It looks weird to have a completely empty sidebar.

Okay, well, there's precedent.

screen shot 2015-02-13 at 4 31 40 pm

**Comment by whit537** [comment]
@Changaco I think we shouldn't block on the empty sidebar, since that's not unique to this page. Do you want to block on pagination?

**Comment by whit537** [comment]
I'm not going to get pagination done today. I think we should abstract out a nice javascript table library like we have for charts, and use that for the teams listing as well as the team members listing and possibly in other places as well.

**Comment by whit537** [comment]
Maybe elsewhere team listings, maybe communities, maybe giving and receiving on profiles. I think building that abstraction should be out of scope here.

**Comment by Changaco** [comment]
>> It should be goal >= 0.
>
>Why?

Because that's how `accepts_tips` is defined.

**Comment by whit537** [comment]
> Because that's how `accepts_tips` is defined.

But why should we use that condition here?

**Comment by Changaco** [comment]
> But why should we use that condition here?

Because the idea was to filter out teams who can't receive, not teams who say they're here to give but can still receive.

**Comment by whit537** [comment]
> Because the idea was to filter out teams who can't receive, not teams who say they're here to give but can still receive.

Done in e9e030305d2898cf25bb75ab4866861f68570668.

**Pull request merged**